core/src/main/scala/kafka/coordinator/transaction/TransactionCoordinator.scala

```diff
@@ -396,8 +397,14 @@ class TransactionCoordinator(txnConfig: TransactionConfig,
           } else if (txnMetadata.producerEpoch != producerEpoch) {
             Left(Errors.PRODUCER_FENCED)
           } else if (txnMetadata.pendingTransitionInProgress) {
-            // return a retriable exception to let the client backoff and retry
-            Left(Errors.CONCURRENT_TRANSACTIONS)
+            // If we are in the produce path, we want to avoid OutOfOrderSequence errors if the added partition is pending.
+            // TODO: Part 2 of KIP-890 will always start transactions with sequence 0, so we enforce that and avoid this workaround.
```

Comment:
We generally avoid `TODO` in the code - they always go stale and give the impression that something is not complete. The comment itself is fine.

Reply:
I can remove the "TODO" parts of the comments.